cpu/atxmega/include/cpu_ebi.h

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (C) 2021 Gerson Fernando Budke
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup         cpu_atxmega
+ * @{
+ *
+ * @file
+ * @brief           External Bus Interface API
+ *
+ * @author          Gerson Fernando Budke <nandojve@gmail.com>
+ */
+
+#include "periph_cpu.h"
+
+#ifndef CPU_EBI_H
+#define CPU_EBI_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef uint32_t hugemem_ptr_t;
+
+#define HUGEMEM_NULL        0
+
+void ebi_init(void);
+
+static inline uint_fast8_t hugemem_read8(const hugemem_ptr_t from)
+{
+    uint8_t value;
+
+    __asm__ volatile (
+        "movw r30, %A1                  \n\t"
+        "out %2, %C1                    \n\t"
+        "ld %0, Z                       \n\t"
+        "out %2, __zero_reg__           \n\t"
+        :
+        "=r"(value)
+        :
+        "r"(from), "i"(&RAMPZ)
+        :
+        "r30", "r31"
+    );
```

Comment:
```suggestion
    __asm__ volatile (
        /* place 24 bit address into r30, r31, RAMPZ */
        "movw   r31:r30,    %B[from]:%A[from]  \n\t"
        "out    %[rampz],   %C[from]           \n\t"
        /* read byte from 24 bit address */
        "ld     %[dest],    Z                  \n\t"
        /* clear ramp
        "out    %[rampz],   __zero_reg__       \n\t"
        : [dest]    "=r"(value)
        : [from]    "r"(from),
          [rampz]   "i"(&RAMPZ)
        : "r30", "r31"
    );
```

IMO it is much easier to read if you use e.g. `%[dest]` instead of `%0`. For `movw`, it can be helpful to be a bit more verbose and write down the whole register pair, as in other assembly dialects `movw` often stands for move-word. It took me a while to remember that `movw` is copying a pair of registers, not a single one.

Are you sure that the constraint `"r"` is sufficient for `from`? If I recall correctly, `movw` only works with the lower register number being even.

Reply:
Agreed, but unfortunately this syntax `movw   r31:r30, %B[from]:%A[from]` don't build.